tests/Fixer/StringNotation/EscapeImplicitBackslashesFixerTest.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\StringNotation;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\StringNotation\EscapeImplicitBackslashesFixer
+ */
+final class EscapeImplicitBackslashesFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     * @param null|array  $configuration
+     *
+     * @dataProvider provideTestFixCases
+     */
+    public function testFix($expected, $input = null, array $configuration = null)
+    {
+        if (null !== $configuration) {
+            $this->fixer->configure($configuration);
+        }
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideTestFixCases()
+    {
+        return [
+            [
+<<<'EOF'
+<?php $var = 'String (\\\'\r\n\x0) for My\Prefix\\';
+EOF
+,
+            ],
+            [
+<<<'EOF'
+<?php $var = 'String (\\\'\\r\\n\\x0) for My\\Prefix\\';
+EOF
+,
+<<<'EOF'
+<?php $var = 'String (\\\'\r\n\x0) for My\Prefix\\';
+EOF
+,
+            ['single_quoted' => true],
+            ],
+            [
+<<<'EOF'
+<?php
+$var = "\\A\\B\\C\\D\\E\\F\\G\\H\\I\\J\\K\\L\\M\\N\\O\\P\\Q\\R\\S\\T\\U\\V\\W\\X\\Y\\Z";
+$var = "\\a\\b\\c\\d\\g\\h\\i\\j\\k\\l\\m\\o\\p\\q\\s\\w\\y\\z \\' \\8\\9 \\xZ \\u";
+$var = <<<HEREDOC_SYNTAX
+\\A\\B\\C\\D\\E\\F\\G\\H\\I\\J\\K\\L\\M\\N\\O\\P\\Q\\R\\S\\T\\U\\V\\W\\X\\Y\\Z
+\\a\\b\\c\\d\\g\\h\\i\\j\\k\\l\\m\\o\\p\\q\\s\\w\\y\\z
+\\'
+\\8\\9
+\\xZ
+\\u
+HEREDOC_SYNTAX;
+$var = <<<'NOWDOC_SYNTAX'
+\A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z
+\a\b\c\d\g\h\i\j\k\l\m\o\p\q\s\w\y\z
+\'
+\8\9
+\xZ
+\u
+NOWDOC_SYNTAX;
+
+EOF
+,
+<<<'EOF'
+<?php
+$var = "\A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z";
+$var = "\a\b\c\d\g\h\i\j\k\l\m\o\p\q\s\w\y\z \' \8\9 \xZ \u";
+$var = <<<HEREDOC_SYNTAX
+\A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z
+\a\b\c\d\g\h\i\j\k\l\m\o\p\q\s\w\y\z
+\'
+\8\9
+\xZ
+\u
+HEREDOC_SYNTAX;
+$var = <<<'NOWDOC_SYNTAX'
+\A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z
+\a\b\c\d\g\h\i\j\k\l\m\o\p\q\s\w\y\z
+\'
+\8\9
+\xZ
+\u
+NOWDOC_SYNTAX;
+
+EOF
+,
+            ],
+            [
+<<<'EOF'
+<?php
+$var = "\e\f\n\r\t\v \\ \$ \"";
+$var = <<<HEREDOC_SYNTAX
+\e\f\n\r\t\v \\ \$ \"
```

Comment:
Double-quotes in HEREDOC syntax do not need to be escaped so the backslash before it has no special meaning and should be doubled.

Reply:
Thank you for the pinpoint, done